class.jetpack-connection-banner.php

```diff
@@ -202,10 +205,19 @@ function render_banner() {
 				</span>
 			</div>
 			<div class="jp-wpcom-connect__inner-container">
-				<span
-					class="notice-dismiss connection-banner-dismiss"
-					title="<?php esc_attr_e( 'Dismiss this notice', 'jetpack' ); ?>">
-				</span>
+
+				<?php
+				if ( ! apply_filters( 'jetpack_pre_connection_prompt_helpers', false ) ) :
```

Comment:
Instead of calling the filter multiple times, I wonder if it would be cleaner to add a new method to the class that would do that check? Something like this:

```php
/**
 * Decide if the site owner should be forced to take action (connect their site or deactivate the plugin)
 * instead of dismissing invitations to connect and leaving a useless unconnected Jetpack plugin on their site.
 *
 * @since 8.6.0
 *
 * @return bool $force_connection_action Default to false.
 */
private function force_connection_action() {
	/**
	 * Allow customization of the behavior of pre-connection banners.
	 *
	 * If this filter is set to true, the site owner must act
	 * and cannot choose to dismiss invitations to connect their site.
	 *
	 * @since 8.6.0
	 *
	 * @param bool $always_show_prompt Should the connection prompts always appear? Default to false.
	 */
	return apply_filters( 'jetpack_pre_connection_force_action', false );
}
```

And then we'd rely on that method everywhere.

Reply:
Done. Thanks for the tip!